### PR TITLE
Ref #5374: Move request blocking stats tracking to `RequestBlockingContentHelper`

### PR DESCRIPTION
**Pull request by cuba**
## Summary of Changes
Continuation of https://github.com/brave/brave-ios/pull/5444

This pull request references #5374

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Review by iccub — APPROVED** [review]
Looks good to me, let's wait for Brandons review before merge

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**
